### ISSUE TITLE
Potential focus visibility issue in annotations/hotspots

### ISSUE THREAD

**Issue by Malvoz** [google/model-viewer#5192]
## Potential a11y issue

Annotations may fail WCAG [Focus Visible](https://www.w3.org/WAI/WCAG22/Understanding/focus-visible.html) and/or [Focus Appearance](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance) on focus.

### Steps to reproduce:

1. Go to https://modelviewer.dev/examples/annotations/#cameraViews
2. Tab to the first annotation/hotspot
3. Notice how the focused annotation element and belonging focus outline is barely visible:

<img width="600" alt="Image" src="https://github.com/user-attachments/assets/5290ea9a-4b61-4995-a6b4-3cd57160c3e5" />

### Relevant CSS:

https://github.com/google/model-viewer/blob/297ed2bdbea0c8f921d985ff0c71afd3a819e12e/packages/model-viewer/src/template.ts#L106-L108

## Potential solution

We can ensure the element and focus outline is visible on focus by not changing the opacity if there's [focus-within](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Selectors/:focus-within) the slotted annotation wrapper:

```css
.annotation-wrapper.hide ::slotted(:not(:focus-within)) {
   opacity: var(--min-hotspot-opacity, 0.25);
}
```

### Result:

The annotation and focus outline is now fully visible if there's focus within:

<img width="600" alt="Image" src="https://github.com/user-attachments/assets/597b75bb-75ba-4dee-9f48-262c6d68cd4a" />

## Disclaimer

I'm not personally a user of `<model-viewer>`, I'm here mostly due to @svinkle's model a11y work and my interest in [the `<model>` element proposal](https://github.com/immersive-web/model-element).
